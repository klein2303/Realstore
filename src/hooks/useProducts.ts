import {useQuery, UseQueryResult} from "@tanstack/react-query";
import {Product} from "../models/Product";

const fetchProducts = async (
    category?: string,
    productId?: number,
    minPrice?: number,
    maxPrice?: number,
    minRating?: number,
): Promise<Product[]> => {
    let url = "https://fakestoreapi.com/products";

    if (productId) {
        url = `${url}/${productId}`;
        const response = await fetch(url);
        if (!response.ok) throw new Error("Error fetching product");
        const product: Product = await response.json();
        return [product]; // Return a single product wrapped in an array
    } else if (category) {
        url = `${url}/category/${category}`;
    }

    const response = await fetch(url);
    if (!response.ok) throw new Error("Error fetching products");
    let products: Product[] = await response.json();

    // Filter products based on price and rating
    if (minPrice !== undefined) {
        products = products.filter((product) => product.price >= minPrice);
    }
    if (maxPrice !== undefined) {
        products = products.filter((product) => product.price <= maxPrice);
    }
    if (minRating !== undefined) {
        products = products.filter((product) => product.rating.rate >= minRating);
    }

    return products;
};

// Custom hook to fetch products with optional filters
export const useProducts = (
    category?: string,
    productId?: number,
    minPrice?: number,
    maxPrice?: number,
    minRating?: number,
): UseQueryResult<Product[]> => {
    const queryKey = ["products", category, productId, minPrice, maxPrice, minRating];

    return useQuery<Product[]>({
        queryKey,
        queryFn: () => fetchProducts(category, productId, minPrice, maxPrice, minRating),
    });
};
