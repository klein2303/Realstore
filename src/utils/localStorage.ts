// Function to store favorite items in local storage
export const setFavorites = (items: number[]): void => {
    localStorage.setItem("favorites", JSON.stringify(items));
};

// Function to get favorite items from local storage
export const getFavorites = (): number[] => {
    const storedItems = localStorage.getItem("favorites");
    return storedItems ? JSON.parse(storedItems) : [];
};

export const toggleFavorite = (productId: number): void => {
    const favorites = getFavorites();
    const isFavorite = favorites.includes(productId);

    const updatedFavorites = isFavorite ? favorites.filter((id) => id !== productId) : [...favorites, productId];

    setFavorites(updatedFavorites);
};
