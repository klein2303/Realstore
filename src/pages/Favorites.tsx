import Navbar from "../components/Navbar/Navbar";
import Footer from "../components/Footer/Footer";
import styles from "../styles/Favorites.module.css";
import {Link} from "react-router-dom";
import ClothingsCards from "../components/Scrolling/ClothingsCards/ClothingsCards";
import {Product} from "../models/Product";
import {useProducts} from "../hooks/useProducts";
import {getFavorites} from "../utils/localStorage";
import {useEffect, useState} from "react";

const Favorites = () => {
    const [favorites, setFavorites] = useState<number[]>([]);
    const [favoriteProducts, setFavoriteProducts] = useState<Product[]>([]);
    const {data: products, isLoading, isError} = useProducts();

    useEffect(() => {
        if (!products) {
            return;
        }
        const favorites = getFavorites();
        setFavorites(favorites);
        const currentFavoriteProducts = products.filter((product: Product) => favorites.includes(product.id));
        setFavoriteProducts(currentFavoriteProducts);
    }, [products]);

    if (isLoading) {
        return (
            <section role="status" aria-live="polite">
                Loading...
            </section>
        );
    }
    if (isError) {
        return (
            <section role="alert" aria-live="assertive">
                Error fetching products.
            </section>
        );
    }

    return (
        <>
            <Navbar />
            <main className={styles.favoritepage} role="region" aria-label="Favorite Products">
                <div className={styles.favoritecontainer}>
                    <nav className={styles.path} aria-label="Navigation">
                        <Link className={styles.pathlink} to={"/"}>
                            Home
                        </Link>{" "}
                        / Favorites
                    </nav>
                    <p className={styles.title}>FAVORITES</p>
                    <section className={styles.favoriteitems} role="list" aria-description="List of favorite products">
                        {favorites.length === 0 ? (
                            <section className={styles.message} role="status" aria-live="polite">
                                No products available.
                            </section>
                        ) : (
                            favoriteProducts.map((product: Product) => (
                                <ClothingsCards
                                    key={product.id}
                                    id={product.id}
                                    title={product.title.substring(0, 20) + "..."}
                                    price={product.price}
                                    category={product.category.charAt(0).toUpperCase() + product.category.substring(1)}
                                    image={product.image}
                                    cart={true}
                                    favorite={true}
                                />
                            ))
                        )}
                    </section>
                </div>
            </main>
            <Footer />
        </>
    );
};

export default Favorites;
