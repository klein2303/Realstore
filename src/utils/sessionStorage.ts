import {Filter} from "../pages/Shopping";

export type CartItem = {productId: number; quantity: number};

// Helper function to store items in session storage
const setSessionStorageItems = <T>(key: string, items: T[]): void => {
    sessionStorage.setItem(key, JSON.stringify(items));
};

// Helper function to get items from session storage
const getSessionStorageItems = <T>(key: string): T[] => {
    const storedItems = sessionStorage.getItem(key);
    return storedItems ? JSON.parse(storedItems) : [];
};

// Cart-specific functions
export const getCart = (): CartItem[] => {
    return getSessionStorageItems<CartItem>("cart");
};

export const addToCart = (productId: number, quantity: number = 1): void => {
    const cart = getCart();
    const existingItem = cart.find((item) => item.productId === productId);

    if (existingItem) {
        existingItem.quantity += quantity; // Increment quantity
    } else {
        cart.push({productId, quantity});
    }

    setSessionStorageItems("cart", cart);
};

export const clearCart = (): void => {
    setSessionStorageItems("cart", []);
};

export const updateCartQuantity = (productId: number, quantity: number): void => {
    const cart = getCart();
    const index = cart.findIndex((item) => item.productId === productId);

    if (index !== -1) {
        // Product exists in the cart
        if (quantity > 0) {
            cart[index].quantity = quantity; // Update quantity
        } else {
            cart.splice(index, 1); // Remove if quantity is 0
        }
    } else if (quantity > 0) {
        // Product doesn't exist in the cart, add it
        cart.push({productId, quantity});
    }

    setSessionStorageItems("cart", cart); // Save updated cart to sessionStorage
};

// Filtered items-specific functions
export const getFilter = (): Filter[] => {
    return getSessionStorageItems<Filter>("filter");
};

export const setFilter = (filter: Filter[]): void => {
    setSessionStorageItems("filter", filter);
};

export const clearFilter = (): void => {
    setSessionStorageItems("filter", []);
};
