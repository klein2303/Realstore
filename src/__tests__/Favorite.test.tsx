import {render, screen} from "@testing-library/react";
import {describe, it, expect, beforeEach, vi} from "vitest";
import Favorites from "../pages/Favorites";
import {useProducts} from "../hooks/useProducts";
import {getFavorites} from "../utils/localStorage";
import {MemoryRouter} from "react-router-dom";

// Mock useProducts and localStorage functions
vi.mock("../hooks/useProducts", () => ({
    useProducts: vi.fn(),
}));

vi.mock("../utils/localStorage", () => ({
    getFavorites: vi.fn(),
    setFavorites: vi.fn(),
    toggleFavorite: vi.fn(),
}));

// Reset mocks before each test
beforeEach(() => {
    vi.resetAllMocks();
    (getFavorites as jest.Mock).mockReturnValue([]);
});

describe("Favorites component", () => {
    it("renders loading state initially", () => {
        // Mock useProducts to simulate the loading state
        (useProducts as jest.Mock).mockReturnValue({
            data: null,
            isLoading: true,
            isError: false,
        });

        const {container} = render(
            <MemoryRouter>
                <Favorites />
            </MemoryRouter>,
        );

        expect(screen.getByText(/Loading.../i)).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("renders error state when data fetching fails", () => {
        // Mock useProducts to simulate an error
        (useProducts as jest.Mock).mockReturnValue({
            data: null,
            isLoading: false,
            isError: true,
        });

        const {container} = render(
            <MemoryRouter>
                <Favorites />
            </MemoryRouter>,
        );

        expect(screen.getByText(/Error fetching products./i)).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("renders favorite products", () => {
        // Mock product data
        const fakeProducts = [
            {
                id: 1,
                title: "Product 1",
                price: 10,
                description: "Description 1",
                category: "category1",
                image: "image1.jpg",
                rating: {
                    rate: 4.5,
                    count: 10,
                },
            },
            {
                id: 2,
                title: "Product 2",
                price: 20,
                description: "Description 2",
                category: "category2",
                image: "image2.jpg",
                rating: {
                    rate: 4.0,
                    count: 20,
                },
            },
        ];

        // Mock useProducts to return the product data
        (useProducts as jest.Mock).mockReturnValue({
            data: fakeProducts,
            isLoading: false,
            isError: false,
        });

        // Mock getFavorites to return product IDs that exist in fakeProducts
        (getFavorites as jest.Mock).mockReturnValue([1, 2]);

        const {container} = render(
            <MemoryRouter>
                <Favorites />
            </MemoryRouter>,
        );

        expect(screen.getByText(/Product 1/i)).toBeInTheDocument();
        expect(screen.getByText(/Product 2/i)).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("renders 'No products available' message when there are no favorite products", () => {
        // Mock useProducts to return an empty product list
        (useProducts as jest.Mock).mockReturnValue({
            data: [],
            isLoading: false,
            isError: false,
        });

        // Mock getFavorites to return an empty list
        (getFavorites as jest.Mock).mockReturnValue([]);

        const {container} = render(
            <MemoryRouter>
                <Favorites />
            </MemoryRouter>,
        );

        expect(screen.getByText(/No products available./i)).toBeInTheDocument(); // Updated text matcher
        expect(container).toMatchSnapshot();
    });
});
