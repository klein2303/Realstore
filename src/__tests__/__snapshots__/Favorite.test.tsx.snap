// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Favorites component > renders 'No products available' message when there are no favorite products 1`] = `
<div>
  <main>
    <nav
      aria-label="Main Navigation"
      class="_navbar_278c00"
      role="navigation"
    >
      <svg
        aria-label="Open Menu"
        class="_hamburgmenu_278c00"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        aria-label="Desktop Navigation Links"
        class="_navpage_278c00"
        role="navigation"
      >
        <a
          class="_navlink_278c00"
          href="/"
        >
          <p>
            HOME
          </p>
        </a>
        <a
          class="_navlink_278c00"
          href="/Shopping"
        >
          <p>
            SHOP
          </p>
        </a>
      </div>
      <div
        class="_navlogo_278c00"
      >
        <a
          href="/"
        >
          <img
            alt="RealStore logo"
            aria-label="RealStore Logo"
            class="_logoimg_278c00"
            src="/project1/Logo.svg"
          />
        </a>
      </div>
      <div
        aria-label="User Navigation"
        class="_navlinks_278c00"
        role="navigation"
      >
        <a
          class="_cartbutton_278c00"
          href="/ShoppingCart"
        >
          <p>
            MY CART
          </p>
          <svg
            aria-hidden="true"
            class="_whiteicon_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 10.5V6a3.75 3.75 0 1 0-7.5 0v4.5m11.356-1.993 1.263 12c.07.665-.45 1.243-1.119 1.243H4.25a1.125 1.125 0 0 1-1.12-1.243l1.264-12A1.125 1.125 0 0 1 5.513 7.5h12.974c.576 0 1.059.435 1.119 1.007ZM8.625 10.5a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm7.5 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </a>
        <a
          class="_navlink_278c00"
          href="/Favorites"
        >
          <svg
            aria-label="View My Favorites"
            class="_heart_278c00"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Heart"
            >
              <path
                d="M12,20.043a.977.977,0,0,1-.7-.288L4.63,13.08A5.343,5.343,0,0,1,6.053,4.513,5.266,5.266,0,0,1,12,5.371a5.272,5.272,0,0,1,5.947-.858A5.343,5.343,0,0,1,19.37,13.08l-6.676,6.675A.977.977,0,0,1,12,20.043ZM8.355,4.963A4.015,4.015,0,0,0,6.511,5.4,4.4,4.4,0,0,0,4.122,8.643a4.345,4.345,0,0,0,1.215,3.73l6.675,6.675,6.651-6.675a4.345,4.345,0,0,0,1.215-3.73A4.4,4.4,0,0,0,17.489,5.4a4.338,4.338,0,0,0-4.968.852h0a.744.744,0,0,1-1.042,0A4.474,4.474,0,0,0,8.355,4.963Z"
              />
            </g>
          </svg>
        </a>
      </div>
      <a
        aria-label="View My Cart"
        class="_shoppingButton_278c00"
        href="/ShoppingCart"
      >
        <svg
          aria-hidden="true"
          class="_whiteicon_278c00"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 10.5V6a3.75 3.75 0 1 0-7.5 0v4.5m11.356-1.993 1.263 12c.07.665-.45 1.243-1.119 1.243H4.25a1.125 1.125 0 0 1-1.12-1.243l1.264-12A1.125 1.125 0 0 1 5.513 7.5h12.974c.576 0 1.059.435 1.119 1.007ZM8.625 10.5a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm7.5 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </a>
    </nav>
  </main>
  <main
    aria-label="Favorite Products"
    class="_favoritepage_b7f171"
    role="region"
  >
    <div
      class="_favoritecontainer_b7f171"
    >
      <nav
        aria-label="Navigation"
        class="_path_b7f171"
      >
        <a
          class="_pathlink_b7f171"
          href="/"
        >
          Home
        </a>
         
        / Favorites
      </nav>
      <p
        class="_title_b7f171"
      >
        FAVORITES
      </p>
      <section
        aria-description="List of favorite products"
        class="_favoriteitems_b7f171"
        role="list"
      >
        <section
          aria-live="polite"
          class="_message_b7f171"
          role="status"
        >
          No products available.
        </section>
      </section>
    </div>
  </main>
  <footer
    aria-label="Website Footer"
    class="_footer_eb428f"
    role="contentinfo"
  >
    <section
      class="_imageholder_eb428f"
    >
      <img
        alt="Decorative footer image"
        class="_image_eb428f"
        src="/project1/footer.svg"
      />
      <p
        class="_title_eb428f"
      >
        EXPLORE OUR CATALOG
      </p>
      <div
        aria-label="Footer Navigation"
        class="_box_eb428f"
        role="region"
      >
        <div
          class="_informationbox_eb428f"
        >
          <p
            class="_informationtitle_eb428f"
          >
            GENERAL
          </p>
          <a
            class="_informationlink_eb428f"
            href="/"
          >
            <p
              aria-label="Home Link"
              class="_informationtext_eb428f"
            >
              Home
            </p>
          </a>
          <a
            class="_informationlink_eb428f"
            href="/Shopping"
          >
            <p
              class="_informationtext_eb428f"
            >
              Shop
            </p>
          </a>
        </div>
        <div
          aria-hidden="true"
          class="_divider_eb428f"
          role="separator"
        />
        <div
          aria-label="Footer Logo and Information"
          class="_logobox_eb428f"
        >
          <a
            href="/"
          >
            <img
              alt="RealStore logo"
              aria-label="RealStore Logo"
              class="_logoimg_eb428f"
              src="/project1/Logo.svg"
            />
          </a>
          <p
            class="_logoinformation_eb428f"
          >
            @2024 RealStore.com
          </p>
        </div>
      </div>
    </section>
  </footer>
</div>
`;

exports[`Favorites component > renders error state when data fetching fails 1`] = `
<div>
  <section
    aria-live="assertive"
    role="alert"
  >
    Error fetching products.
  </section>
</div>
`;

exports[`Favorites component > renders favorite products 1`] = `
<div>
  <main>
    <nav
      aria-label="Main Navigation"
      class="_navbar_278c00"
      role="navigation"
    >
      <svg
        aria-label="Open Menu"
        class="_hamburgmenu_278c00"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        aria-label="Desktop Navigation Links"
        class="_navpage_278c00"
        role="navigation"
      >
        <a
          class="_navlink_278c00"
          href="/"
        >
          <p>
            HOME
          </p>
        </a>
        <a
          class="_navlink_278c00"
          href="/Shopping"
        >
          <p>
            SHOP
          </p>
        </a>
      </div>
      <div
        class="_navlogo_278c00"
      >
        <a
          href="/"
        >
          <img
            alt="RealStore logo"
            aria-label="RealStore Logo"
            class="_logoimg_278c00"
            src="/project1/Logo.svg"
          />
        </a>
      </div>
      <div
        aria-label="User Navigation"
        class="_navlinks_278c00"
        role="navigation"
      >
        <a
          class="_cartbutton_278c00"
          href="/ShoppingCart"
        >
          <p>
            MY CART
          </p>
          <svg
            aria-hidden="true"
            class="_whiteicon_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 10.5V6a3.75 3.75 0 1 0-7.5 0v4.5m11.356-1.993 1.263 12c.07.665-.45 1.243-1.119 1.243H4.25a1.125 1.125 0 0 1-1.12-1.243l1.264-12A1.125 1.125 0 0 1 5.513 7.5h12.974c.576 0 1.059.435 1.119 1.007ZM8.625 10.5a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm7.5 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </a>
        <a
          class="_navlink_278c00"
          href="/Favorites"
        >
          <svg
            aria-label="View My Favorites"
            class="_heart_278c00"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Heart"
            >
              <path
                d="M12,20.043a.977.977,0,0,1-.7-.288L4.63,13.08A5.343,5.343,0,0,1,6.053,4.513,5.266,5.266,0,0,1,12,5.371a5.272,5.272,0,0,1,5.947-.858A5.343,5.343,0,0,1,19.37,13.08l-6.676,6.675A.977.977,0,0,1,12,20.043ZM8.355,4.963A4.015,4.015,0,0,0,6.511,5.4,4.4,4.4,0,0,0,4.122,8.643a4.345,4.345,0,0,0,1.215,3.73l6.675,6.675,6.651-6.675a4.345,4.345,0,0,0,1.215-3.73A4.4,4.4,0,0,0,17.489,5.4a4.338,4.338,0,0,0-4.968.852h0a.744.744,0,0,1-1.042,0A4.474,4.474,0,0,0,8.355,4.963Z"
              />
            </g>
          </svg>
        </a>
      </div>
      <a
        aria-label="View My Cart"
        class="_shoppingButton_278c00"
        href="/ShoppingCart"
      >
        <svg
          aria-hidden="true"
          class="_whiteicon_278c00"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 10.5V6a3.75 3.75 0 1 0-7.5 0v4.5m11.356-1.993 1.263 12c.07.665-.45 1.243-1.119 1.243H4.25a1.125 1.125 0 0 1-1.12-1.243l1.264-12A1.125 1.125 0 0 1 5.513 7.5h12.974c.576 0 1.059.435 1.119 1.007ZM8.625 10.5a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm7.5 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </a>
    </nav>
  </main>
  <main
    aria-label="Favorite Products"
    class="_favoritepage_b7f171"
    role="region"
  >
    <div
      class="_favoritecontainer_b7f171"
    >
      <nav
        aria-label="Navigation"
        class="_path_b7f171"
      >
        <a
          class="_pathlink_b7f171"
          href="/"
        >
          Home
        </a>
         
        / Favorites
      </nav>
      <p
        class="_title_b7f171"
      >
        FAVORITES
      </p>
      <section
        aria-description="List of favorite products"
        class="_favoriteitems_b7f171"
        role="list"
      >
        <article
          aria-label="Product card for Product 1..."
          class="_clothingSection_47ab48"
          role="region"
        >
          <figure
            class="_imgContainer_47ab48"
          >
            <a
              href="/Clothe/1"
            >
              <img
                alt="Image of Product 1..."
                class="_clothingImage_47ab48"
                src="image1.jpg"
              />
            </a>
          </figure>
          <section
            class="_descriptionSpace_47ab48"
          >
            <header
              class="_productType_47ab48"
            >
              <h5
                class="_category_47ab48"
              >
                Category1
              </h5>
              <section
                aria-label="Product Actions"
                class="_IconSpace_47ab48"
              >
                <span
                  aria-label="Add to cart"
                  role="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    tabindex="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="176"
                      cy="416"
                      fill="none"
                      r="16"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="32"
                    />
                    <circle
                      cx="400"
                      cy="416"
                      fill="none"
                      r="16"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="32"
                    />
                    <path
                      d="M48 80h64l48 272h256"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="32"
                    />
                    <path
                      d="M160 288h249.44a8 8 0 0 0 7.85-6.43l28.8-144a8 8 0 0 0-7.85-9.57H128"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="32"
                    />
                  </svg>
                </span>
                <span
                  aria-label="Remove from favorites"
                  role="button"
                >
                  <svg
                    class="_favorited_47ab48"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    tabindex="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 448l-30.164-27.211C118.718 322.442 48 258.61 48 179.095 48 114.221 97.918 64 162.4 64c36.399 0 70.717 16.742 93.6 43.947C278.882 80.742 313.199 64 349.6 64 414.082 64 464 114.221 464 179.095c0 79.516-70.719 143.348-177.836 241.694L256 448z"
                    />
                  </svg>
                </span>
              </section>
            </header>
            <footer
              class="_nameAndPrice_47ab48"
            >
              <span
                class="_productname_47ab48"
              >
                <h4
                  class="_h4text_47ab48"
                >
                  Product 1...
                </h4>
              </span>
              <span
                class="_price_47ab48"
              >
                <h4
                  class="_h4text_47ab48"
                >
                  $ 
                  10
                </h4>
              </span>
            </footer>
          </section>
        </article>
        <article
          aria-label="Product card for Product 2..."
          class="_clothingSection_47ab48"
          role="region"
        >
          <figure
            class="_imgContainer_47ab48"
          >
            <a
              href="/Clothe/2"
            >
              <img
                alt="Image of Product 2..."
                class="_clothingImage_47ab48"
                src="image2.jpg"
              />
            </a>
          </figure>
          <section
            class="_descriptionSpace_47ab48"
          >
            <header
              class="_productType_47ab48"
            >
              <h5
                class="_category_47ab48"
              >
                Category2
              </h5>
              <section
                aria-label="Product Actions"
                class="_IconSpace_47ab48"
              >
                <span
                  aria-label="Add to cart"
                  role="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    tabindex="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="176"
                      cy="416"
                      fill="none"
                      r="16"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="32"
                    />
                    <circle
                      cx="400"
                      cy="416"
                      fill="none"
                      r="16"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="32"
                    />
                    <path
                      d="M48 80h64l48 272h256"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="32"
                    />
                    <path
                      d="M160 288h249.44a8 8 0 0 0 7.85-6.43l28.8-144a8 8 0 0 0-7.85-9.57H128"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="32"
                    />
                  </svg>
                </span>
                <span
                  aria-label="Remove from favorites"
                  role="button"
                >
                  <svg
                    class="_favorited_47ab48"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    tabindex="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 448l-30.164-27.211C118.718 322.442 48 258.61 48 179.095 48 114.221 97.918 64 162.4 64c36.399 0 70.717 16.742 93.6 43.947C278.882 80.742 313.199 64 349.6 64 414.082 64 464 114.221 464 179.095c0 79.516-70.719 143.348-177.836 241.694L256 448z"
                    />
                  </svg>
                </span>
              </section>
            </header>
            <footer
              class="_nameAndPrice_47ab48"
            >
              <span
                class="_productname_47ab48"
              >
                <h4
                  class="_h4text_47ab48"
                >
                  Product 2...
                </h4>
              </span>
              <span
                class="_price_47ab48"
              >
                <h4
                  class="_h4text_47ab48"
                >
                  $ 
                  20
                </h4>
              </span>
            </footer>
          </section>
        </article>
      </section>
    </div>
  </main>
  <footer
    aria-label="Website Footer"
    class="_footer_eb428f"
    role="contentinfo"
  >
    <section
      class="_imageholder_eb428f"
    >
      <img
        alt="Decorative footer image"
        class="_image_eb428f"
        src="/project1/footer.svg"
      />
      <p
        class="_title_eb428f"
      >
        EXPLORE OUR CATALOG
      </p>
      <div
        aria-label="Footer Navigation"
        class="_box_eb428f"
        role="region"
      >
        <div
          class="_informationbox_eb428f"
        >
          <p
            class="_informationtitle_eb428f"
          >
            GENERAL
          </p>
          <a
            class="_informationlink_eb428f"
            href="/"
          >
            <p
              aria-label="Home Link"
              class="_informationtext_eb428f"
            >
              Home
            </p>
          </a>
          <a
            class="_informationlink_eb428f"
            href="/Shopping"
          >
            <p
              class="_informationtext_eb428f"
            >
              Shop
            </p>
          </a>
        </div>
        <div
          aria-hidden="true"
          class="_divider_eb428f"
          role="separator"
        />
        <div
          aria-label="Footer Logo and Information"
          class="_logobox_eb428f"
        >
          <a
            href="/"
          >
            <img
              alt="RealStore logo"
              aria-label="RealStore Logo"
              class="_logoimg_eb428f"
              src="/project1/Logo.svg"
            />
          </a>
          <p
            class="_logoinformation_eb428f"
          >
            @2024 RealStore.com
          </p>
        </div>
      </div>
    </section>
  </footer>
</div>
`;

exports[`Favorites component > renders loading state initially 1`] = `
<div>
  <section
    aria-live="polite"
    role="status"
  >
    Loading...
  </section>
</div>
`;
